code/src/containers/forms/delivery-form/DeliveryForm.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { act, fireEvent, screen, waitFor } from "@testing-library/react";
+
+import DeliveryForm from "@/containers/forms/delivery-form/DeliveryForm";
+
+import useCreateOrder from "@/hooks/use-create-order/useCreateOrder";
+import useGetUserDetails from "@/hooks/use-get-user-details/useGetUserDetails";
+import renderWithProviders from "@/utils/render-with-providers/renderWithProviders";
+
+jest.mock("@/hooks/use-create-order/useCreateOrder", () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
+
+jest.mock("@/hooks/use-get-user-details/useGetUserDetails", () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
+
+const mockCreateOrder = jest.fn();
+const mockUserDetails = {
+  id: 123,
+  firstName: "John",
+  lastName: "Doe",
+  email: "john.doe@example.com"
+};
+
+describe("DeliveryForm", () => {
+  beforeEach(() => {
+    (useCreateOrder as jest.Mock).mockReturnValue([
+      mockCreateOrder,
+      { isLoading: false }
+    ]);
+    (useGetUserDetails as jest.Mock).mockReturnValue(mockUserDetails);
+    renderWithProviders(<DeliveryForm totalPrice={100} />);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("renders the delivery form correctly", () => {
+    const cityInput = screen.getByLabelText(/deliveryForm.city/);
+    const departmentInput = screen.getByLabelText(/deliveryForm.department/);
+    const deliveryMethodSelect = screen.getByLabelText(
+      /deliveryForm.postMethod/
+    );
+    const submitButton = screen.getByRole("button");
+
+    expect(cityInput).toBeInTheDocument();
+    expect(departmentInput).toBeInTheDocument();
+    expect(deliveryMethodSelect).toBeInTheDocument();
+    expect(submitButton).toBeInTheDocument();
+  });
+
+  test("handles input changes and form submission", async () => {
+    const cityInput = screen.getByLabelText(/deliveryForm.city/);
+    const departmentInput = screen.getByLabelText(/deliveryForm.department/);
+
+    const submitButton = screen.getByRole("button");
+
+    await act(() => {
+      fireEvent.change(cityInput, { target: { value: "New York" } });
+      fireEvent.change(departmentInput, { target: { value: "123" } });
+    });
+
+    fireEvent.click(submitButton);
+
+    await waitFor(() => {
```

Comment:
Do we need `waitFor` here?

Reply:
Yes
![image](https://github.com/user-attachments/assets/c9f185bf-441b-4a29-af5b-dd0d63f03c1b)